Dockerfile.test

```diff
@@ -0,0 +1,41 @@
+# Dockerfile used for running GPU tests in self-hosted GitHub Actions runner.
+
+FROM python:3.7
+
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+
+ENV PATH /usr/local/nvidia/bin/:$PATH
+ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
+
+# Tell nvidia-docker the driver spec that we need as well as to
+# use all available devices, which are mounted at /usr/local/nvidia.
+# The LABEL supports an older version of nvidia-docker, the env
+# variables a newer one.
+ENV NVIDIA_VISIBLE_DEVICES all
+ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
+LABEL com.nvidia.volumes.needed="nvidia_driver"
+
+WORKDIR /stage/allennlp
+
+# Add minimal files necessary to install dependencies.
```

Comment:
Why? So that we have a docker layer that captures all dependencies?

If that's the case, can we also use this trick in the other Dockerfiles?